core/src/main/java/com/linecorp/armeria/common/TlsProvider.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Consumer;
+
+import com.linecorp.armeria.client.ClientTlsProviderBuilder;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.common.metric.MeterIdPrefix;
+import com.linecorp.armeria.common.metric.MoreMeterBinders;
+import com.linecorp.armeria.server.ServerTlsProviderBuilder;
+
+import io.netty.handler.ssl.SslContextBuilder;
+
+/**
+ * Provides {@link TlsKeyPair}s for TLS handshakes.
+ */
+@UnstableApi
+@FunctionalInterface
+public interface TlsProvider {
```

Comment:
Do we need `TlsProvider.of{Client,Server}(TlsKeyPair)` shortcut as well?

Reply:
I thought `TlsProvider.of()` was sufficient for now, but considering the future, it would be better to split it into server and client.